app/models/rental.rb

```diff
@@ -1,15 +1,30 @@
 class Rental < ApplicationRecord
   belongs_to :user
   belongs_to :product
+  before_save :generate_confirmation_token
```

Comment:
It's better to use before_validate callback here. In some cases, token can be invalid(in a future)